### PR TITLE
DF-193 Add BFS to ISCO08 occupation mapping

### PR DESCRIPTION
**Pull request by varusakov**

### CODE REVIEW

File: src/main/java/ch/admin/seco/service/reference/domain/OccupationLabelMappingISCO.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    @GenericGenerator(name = "UUIDGenerator", strategy = "uuid2")
+    private UUID id;
+
+    @NaturalId
+    @NotNull
+    @Column(name = "bfs_code", nullable = false, unique = true)
+    private String bfsCode;
+
+    @NotNull
+    @Column(name = "isco_08_code", nullable = false)
```
Comment (cjpsycos) [inline src/main/java/ch/admin/seco/service/reference/domain/OccupationLabelMappingISCO.java]: Why is the name of the column different to the field?
For what stays 08?

File: src/main/java/ch/admin/seco/service/reference/service/impl/OccupationLabelServiceImpl.java [hunk 2]
```diff
@@ -67,8 +73,24 @@ public OccupationLabel save(OccupationLabel occupationLabel) {
...
         log.debug("Request to get OccupationLabelMapping : {}", professionCode);
+        return findOccupationMapping(professionCode).map(occupationLabelMapping -> {
+            OccupationLabelMappingDto dto = new OccupationLabelMappingDto()
+                .id(occupationLabelMapping.getId())
+                .bfsCode(occupationLabelMapping.getBfsCode())
+                .avamCode(occupationLabelMapping.getAvamCode())
+                .sbn3Code(occupationLabelMapping.getSbn3Code())
+                .sbn5Code(occupationLabelMapping.getSbn5Code())
+                .description(occupationLabelMapping.getDescription());
+            occupationLabelMappingISCORepository.findOneByBfsCode(occupationLabelMapping.getBfsCode())
```
Comment (cjpsycos) [inline src/main/java/ch/admin/seco/service/reference/service/impl/OccupationLabelServiceImpl.java]: I would get first the iscoCode and then create the dto and return it

File: src/main/java/ch/admin/seco/service/reference/domain/OccupationLabelMappingISCO.java (hunk 1 above)
Comment (varusakov) [reply src/main/java/ch/admin/seco/service/reference/domain/OccupationLabelMappingISCO.java]: It's full classification system name. Renamed to simple `isco_code`

File: src/main/java/ch/admin/seco/service/reference/service/impl/OccupationLabelServiceImpl.java (hunk 2 above)
Comment (varusakov) [reply src/main/java/ch/admin/seco/service/reference/service/impl/OccupationLabelServiceImpl.java]: Tnx, updated this logic

File: src/main/java/ch/admin/seco/service/reference/service/impl/OccupationLabelServiceImpl.java [hunk 3]
```diff
@@ -67,8 +73,23 @@ public OccupationLabel save(OccupationLabel occupationLabel) {
     }
 
     @Override
-    public Optional<OccupationLabelMapping> findOneOccupationMapping(ProfessionCodeDTO professionCode) {
+    public Optional<OccupationLabelMappingDto> findOneOccupationMapping(ProfessionCodeDTO professionCode) {
         log.debug("Request to get OccupationLabelMapping : {}", professionCode);
+        return findOccupationMapping(professionCode).map(occupationLabelMapping -> {
+            String iscoCode = occupationLabelMappingISCORepository.findOneByBfsCode(occupationLabelMapping.getBfsCode())
```
Comment (birom) [inline src/main/java/ch/admin/seco/service/reference/service/impl/OccupationLabelServiceImpl.java]: I think we need to check if the mapping has a BSF code. I case of x28 it is not guaranteed.

**Pull request merged**